kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -387,6 +397,66 @@ private synchronized void manageNonConfiguredInterfaces(List<String> configuredI
         }
     }
 
+    private void disable(Device device) throws DBusException {
+        NMDeviceState deviceState = getDeviceState(device);
+        if (Boolean.TRUE.equals(NMDeviceState.isConnected(deviceState))) {
+            DeviceStateLock dsLock = new DeviceStateLock(this.dbusConnection, device.getObjectPath(),
+                    NMDeviceState.NM_DEVICE_STATE_DISCONNECTED);
+            device.Disconnect();
+            dsLock.waitForSignal();
+        }
+
+        Optional<Connection> connection = getAppliedConnection(device);
+        if (connection.isPresent()) {
+            connection.get().Delete();
+        }
+
+        NMDeviceType deviceType = getDeviceType(device);
+        if (deviceType == NMDeviceType.NM_DEVICE_TYPE_MODEM) {
+            handleModemManagerGPSSetup(device, Optional.of(false));
+        }
+
+    }
+
+    private void handleModemManagerGPSSetup(Device device, Optional<Boolean> enableGPS) throws DBusException {
+        Optional<String> modemDevicePath = getModemPathFromMM(device.getObjectPath());
+
+        if (!modemDevicePath.isPresent()) {
+            logger.warn("Cannot retrieve MM.Modem from NM.Modem at path: {}. Skipping GPS configuration.",
+                    device.getObjectPath());
+            return;
+        }
+
+        boolean isGPSSourceEnabled = enableGPS.isPresent() && enableGPS.get();
+
+        Location modemLocation = this.dbusConnection.getRemoteObject(MM_BUS_NAME, modemDevicePath.get(),
+                Location.class);
+        Properties modemLocationProperties = this.dbusConnection.getRemoteObject(MM_BUS_NAME,
+                modemLocation.getObjectPath(), Properties.class);
+
+        Set<MMModemLocationSource> availableLocationSources = MMModemLocationSource
+                .toMMModemLocationSourceFromBitMask(modemLocationProperties.Get(MM_LOCATION_BUS_NAME, "Capabilities"));
+        Set<MMModemLocationSource> enabledLocationSources = MMModemLocationSource
+                .toMMModemLocationSourceFromBitMask(modemLocationProperties.Get(MM_LOCATION_BUS_NAME, "Enabled"));
+
+        EnumSet<MMModemLocationSource> desiredLocationSources = EnumSet.noneOf(MMModemLocationSource.class);
+        if (availableLocationSources.contains(MMModemLocationSource.MM_MODEM_LOCATION_SOURCE_GPS_RAW)
+                && isGPSSourceEnabled) {
+            desiredLocationSources.add(MMModemLocationSource.MM_MODEM_LOCATION_SOURCE_GPS_RAW);
+        }
+        if (availableLocationSources.contains(MMModemLocationSource.MM_MODEM_LOCATION_SOURCE_GPS_NMEA)
+                && isGPSSourceEnabled) {
+            desiredLocationSources.add(MMModemLocationSource.MM_MODEM_LOCATION_SOURCE_GPS_NMEA);
+        }
+        if (enabledLocationSources.contains(MMModemLocationSource.MM_MODEM_LOCATION_SOURCE_3GPP_LAC_CI)) {
+            desiredLocationSources.add(MMModemLocationSource.MM_MODEM_LOCATION_SOURCE_3GPP_LAC_CI);
+        }
```

Comment:
What happens if other types of location sources are enabled? It could be better to start from the "Enabled" list and add/removed NMEA sources?

Reply:
Fixed in ccde4d9a8d219d5844a9350735fe8d5f3f1fccbe